packages/xo-web/src/xo-app/settings/remotes/remote.js

```diff
@@ -56,12 +56,20 @@ export default decorate([
           name,
           options = remote.options || '',
           password = remote.password,
-          path = remote.path,
           port = remote.port,
           proxyId = remote.proxy,
           type = remote.type,
           username = remote.username,
         } = state
+        let { path = remote.path } = state
+        if (type === 's3') {
+          const {
+            parsedPath,
+            bucket = parsedPath != null && parsedPath.split('/')[0],
+            directory = parsedPath != null && parsedPath.split('/')[1],
```

Comment:
Why do you split the path to concatenate it again?

Reply:
The intent is to split the path, receive the new values from the state and re-create the path with the updated values.

Reply:
I still don't understand why splitting and then joining, are we normalizing something?

Why not simply `path = parsedPath`?

Reply:
we parse the stored URL, then we give the system the opportunity to update `bucket` and `directory` components from the UI, then we rebuild the path with the maybe updated components